test/sockets.test.js

```diff
@@ -1,63 +1,186 @@
 import 'regenerator-runtime/runtime';
 
 import Socket from '../src/sockets';
-import Logger from '../src/logger';
+import { Logger, SOCKET_PING } from 'syft-helpers.js';
 
 import { WebSocket, Server } from 'mock-socket';
 
 global.WebSocket = WebSocket;
 
 const url = 'ws://localhost:8080/';
 
-const onMessage = event => console.log('NEW MESSAGE', event);
-const onOpen = event => console.log('NEW OPEN', event);
-const onClose = event => console.log('NEW CLOSE', event);
-const onError = event => console.log('NEW ERROR', event);
+/**
+ * Creates promise from the first event call
+ * @param emitter Object that emits the event
+ * @param event Event name
+ * @returns {Promise}
+ */
+function makeEventPromise(emitter, event) {
+  let resolver;
+  let promise = new Promise(resolve => resolver = resolve);
+  emitter.on(event, data => resolver(data));
+  return promise;
+}
 
 describe('Sockets', () => {
+  // common test objects
+  let mockServer;
+
   jest.spyOn(console, 'log');
 
+  beforeEach(() => {
+    mockServer = new Server(url);
+    mockServer.connected = makeEventPromise(mockServer, 'connection');
+  });
+
   afterEach(() => {
+    mockServer.close();
     jest.clearAllMocks();
   });
 
-  test('can communicate with a socket server', async () => {
-    const message = { message: 'This is a test' };
+  test('keep-alive messages', async() => {
+    const keepAliveTimeout = 300;
+    const expectedMessagesCount = 3;
 
-    const mockServer = new Server(url);
     const mySocket = new Socket({
       url,
       logger: new Logger(true),
-      onMessage: event => onMessage(event),
-      onOpen: event => onOpen(event),
-      onClose: event => onClose(event),
-      onError: event => onError(event)
+      keepAliveTimeout
     });
+    let serverSocket = await mockServer.connected;
+    // collect messages
+    let messages = [];
+    serverSocket.on('message', message => messages.push(JSON.parse(message)));
+    // if we wait long enough, we should see `expectedMessagesCount` keep-alive messages
+    await new Promise(done => setTimeout(done,
+      keepAliveTimeout * expectedMessagesCount + keepAliveTimeout / 2));
+    // one message is sent right after establishing the connection hence +1 
+    // (do we really need that?)
+    expect(messages).toHaveLength(expectedMessagesCount + 1);
+    let expectedTypes  =[];
```

Comment:
This spacing is incorrect.  Is prettier running for you?  Normally [Prettier](https://prettier.io/) will fix things like this.

Reply:
I thought prettier would run automatically :) Fixed